tree/dataframe/inc/ROOT/RDF/ColumnReaders.hxx

```diff
@@ -0,0 +1,331 @@
+   // See https://github.com/root-project/root/commit/26e8ace6e47de6794ac9ec770c3bbff9b7f2e945
+   void Reset() final { fTreeValue.reset(); }
+};
+
+/// RTreeColumnReader specialization for TTree values read via TTreeReaderArrays
+///
+/// TTreeReaderArrays are used whenever the RDF column type is RVec<T>.
+template <typename T>
+class R__CLING_PTRCHECK(off) RTreeColumnReader<RVec<T>> final : public RColumnReaderBase<RVec<T>> {
+   std::unique_ptr<TTreeReaderArray<T>> fTreeArray;
+
+   /// Enumerator for the memory layout of the branch
+   enum class EStorageType : char { kContiguous, kUnknown, kSparse };
+
+   /// Signal whether we ever checked that the branch we are reading with a TTreeReaderArray stores array elements
+   /// in contiguous memory.
+   EStorageType fStorageType = EStorageType::kUnknown;
+
+   /// We return a reference to this RVec to clients, to guarantee a stable address and contiguous memory layout.
+   RVec<T> fRVec;
+
+   /// Whether we already printed a warning about performing a copy of the TTreeReaderArray contents
+   bool fCopyWarningPrinted = false;
+
+public:
+   RTreeColumnReader(TTreeReader &r, const std::string &colName)
+      : fTreeArray(std::make_unique<TTreeReaderArray<T>>(r, colName.c_str()))
+   {
+   }
+
+   RVec<T> &Get(Long64_t) final
+   {
+      auto &readerArray = *fTreeArray;
+      // We only use TTreeReaderArrays to read columns that users flagged as type `RVec`, so we need to check
+      // that the branch stores the array as contiguous memory that we can actually wrap in an `RVec`.
+      // Currently we need the first entry to have been loaded to perform the check
+      // TODO Move check to constructor once ROOT-10823 is fixed and TTreeReaderArray itself exposes this information
+      const auto arrSize = readerArray.GetSize();
+      if (EStorageType::kUnknown == fStorageType && arrSize > 1) {
+         // We can decide since the array is long enough
+         fStorageType = EStorageType::kContiguous;
+         for (auto i = 0u; i < arrSize - 1; ++i) {
+            if ((char *)&readerArray[i + 1] - (char *)&readerArray[i] != sizeof(T)) {
+               fStorageType = EStorageType::kSparse;
+               break;
+            }
+         }
+      }
+
+      const auto readerArraySize = readerArray.GetSize();
+      if (EStorageType::kContiguous == fStorageType ||
+          (EStorageType::kUnknown == fStorageType && readerArray.GetSize() < 2)) {
+         if (readerArraySize > 0) {
+            // trigger loading of the contents of the TTreeReaderArray
+            // the address of the first element in the reader array is not necessarily equal to
+            // the address returned by the GetAddress method
+            auto readerArrayAddr = &readerArray.At(0);
+            RVec<T> rvec(readerArrayAddr, readerArraySize);
+            std::swap(fRVec, rvec);
+         } else {
+            RVec<T> emptyVec{};
+            std::swap(fRVec, emptyVec);
```

Comment:
```suggestion
            fRVec.clear();
```

Reply:
they are not equivalent. `clear` does not reduce the vector's capacity, `swap` with an empty vector does. I am not sure we want to keep all `fRVec` data members to the maximum size they need through the event loop, it might cost some ram.